### PR TITLE
GCS server error info handler use storage instead of redis accessor

### PR DESCRIPTION
**Pull request by ffbin**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Review by raulchen — APPROVED** [review]

**Comment by raulchen** [comment]
test failures look related. Can you take a look?

File: src/ray/protobuf/gcs_service.proto [hunk 1]
```diff
@@ -404,10 +404,33 @@ message ReportJobErrorReply {
   GcsStatus status = 1;
 }
 
+message GetAllJobErrorInfoRequest {
+}
+
+message GetAllJobErrorInfoReply {
+  GcsStatus status = 1;
+  // Data of job error.
```
Comment (ashione) [inline src/ray/protobuf/gcs_service.proto:412]:
```suggestion
  // A list of of job error info.
```

File: src/ray/gcs/gcs_server/error_info_handler_impl.cc [hunk 2]
```diff
@@ -23,23 +24,54 @@ void DefaultErrorInfoHandler::HandleReportJobError(
...
-                     << ", type = " << type;
-    }
-    GCS_RPC_SEND_REPLY(send_reply_callback, reply, status);
-  };
-
-  Status status = gcs_client_.Errors().AsyncReportJobError(error_table_data, on_done);
-  if (!status.ok()) {
-    on_done(status);
+  auto &list = job_to_errors_[job_id];
+  if (list.size() >= RayConfig::instance().error_info_entries_size_per_job()) {
```
Comment (ashione) [inline src/ray/gcs/gcs_server/error_info_handler_impl.cc:28]: Maybe boost::circle_buffer is better if we should keep fixed length of data.

**Comment by rkooo567** [comment]
@raulchen @ffbin We don't need this PR anymore right? We will just remove error tables.

**Comment by ffbin** [comment]
We will remove error table,  error pub-sub will be implemented in this pr: https://github.com/ray-project/ray/pull/9665

**Pull request closed without merging**